File: Source/Samples/95_Periodic/code_acontainer.cpp
// AContainer application source code
#include "PeriodicApp.h"

void PeriodicApp::setup_acontainer( AWidget *layout )
{
    PODVector<AWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleAcontainerEvent ));
}

void PeriodicApp::HandleAcontainerEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    AWidget* widget = static_cast<AWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uicontainercode" )
        {
            AppLog( "AContainer support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Sources/code_acontainer.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uicontainerlayout" )
        {
            AppLog( "AContainer support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_acontainer.ui.txt", widget->GetParent() );
        }
    }
}

File: bin/Data/AUI/Periodic/LuaScripts/code_acontainer.lua
-- AContainer application source code

function setup_uicontainer( layout )
    local dest = layout:SearchAWidgetClass( "TBButton" )
    for i, awid in ipairs(dest) do
        SubscribeToEvent(awid, "WidgetEvent", "HandleUicontainerEvent")
	end
end

function HandleUicontainerEvent( eventType, eventData )
    local widget = eventData["Target"]:GetPtr("AWidget")
    if widget == nil then
 		return
	end
    if eventData["Type"]:GetInt() == UI_EVENT_TYPE_CLICK then
        if widget:GetId() == "uicontainercode" then
            AppLog( "AContainer support : " .. widget:GetId() .. " was pressed " )
            ViewCode ( "LuaScripts/code_acontainer.lua", widget:GetParent())
        end
        if widget:GetId() ==  "uicontainerlayout" then
            AppLog( "AContainer support : " .. widget:GetId() .. " was pressed ")
            ViewCode ( "Scenes/layout_acontainer.ui.txt", widget:GetParent())
        end
    end
end

